site/src/components/introduction/building-a-chart.md

```diff
@@ -0,0 +1,188 @@
+
+At this point you might like to read more about the [D3 component pattern](http://bost.ocks.org/mike/chart/) that we follow with d3fc.
+
+I'll forgive you if you are not too impressed with the above example. So far you've got a bar series, but no axes or labels, yet the example involves quite a bit of code.
+
+The reason for this is that the bar series component is relatively low-level component, it renders a series and nothing more. (It's actually not the lowest level of component, the bar series uses the d3fc bar SVG component). In order to quickly assemble a chart you need to make use of a more high-level component ... a chart.
+
+## Cartesian chart
+
+The cartesian chart component does a number of things, it measures its container in order to fill it, setting a suitable range on the scales that you supply (it's still your job to set the domain on each). The chart renders a pair of axes and, optionally, other cosmetic features such as chart and axis labels.
+
+You can render a series within the chart by setting it as the `plotArea`. The chart sets the x and y scales on the component you supply.
+
+Here's the complete example:
+
+```js
+{{{ codeblock cartesian-chart-js }}}
+```
+
+This renders a more complete chart:
+
+{{{ dynamic-include 'codepen' html="cartesian-chart-html" js="cartesian-chart-js"}}}
+{{{ cartesian-chart-html }}}
+
+{{{ dynamic-include 'javascript' js="cartesian-chart-js" }}}
+
+## Setting the axis domain
+
+The current y-domain is a bit of a guess, being hard-coded to a range `[0, 20]`. Ideally the domain should be computed based on the data. In this case the domain should accommodate the greatest sales value, plus some padding, and also extent to zero.
+
+There's another d3fc component that can help here, the [extent](/components/util/extent.html) utility. The following code ensures that zero is included, and adds a small amount of padding to the 'top' end of the range:
+
+```js
+{{{ codeblock cartesian-chart-extent-js }}}
+```
+
+Here's the overall result:
+
+{{{ dynamic-include 'codepen' html="cartesian-chart-extent-html" js="cartesian-chart-extent-js"}}}
+{{{ cartesian-chart-extent-html }}}
+{{{ dynamic-include 'javascript' js="cartesian-chart-extent-js" }}}
+
+## Tidying the chart
+
+Let't tidy up the chart a bit by setting suitable margins and labels:
+
+```js
+{{{ codeblock tidy-chart-js }}}
+```
+
+This gives a much more appealing chart:
+
+{{{ dynamic-include 'codepen' html="tidy-chart-html" js="tidy-chart-js"}}}
+{{{ tidy-chart-html }}}
+{{{ dynamic-include 'javascript' js="tidy-chart-js" }}}
+
+As you can see from the above code, the cartesian chart exposes a number of the properties of the components it encapsulates. Most of the x and y scale and axes properties are exposed, with a suitable `x` or `y` prefix. This makes use of a standard D3 concept of property [rebinding](https://github.com/mbostock/d3/wiki/Internals#rebind), which gives rise to a compositional pattern rather than inheritance.
```

Comment:
This description is very good but the x/y prefix stuff is introduced a few chapters above